### PR TITLE
Fix buggy docs tests

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR fixes the call to `setup_class` in some tests. Also, extends a test for context.

## Fixes

-

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: tests/test_docs/test_agent_vs_aea/test_agent_vs_aea.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ class TestAgentVsAEA(BaseAEATestCase):
     @classmethod
     def setup_class(cls):
         """Setup the test class."""
-        BaseAEATestCase.setup_class()
```
Comment (DavidMinarsch) [inline tests/test_docs/test_agent_vs_aea/test_agent_vs_aea.py:43]: Note, we need to be careful not to do this.

Comment (marcofavorito) [reply tests/test_docs/test_agent_vs_aea/test_agent_vs_aea.py:43]: yes!

**Review by solarw — APPROVED** [review]
LGTM

File: tests/test_docs/test_agent_vs_aea/test_agent_vs_aea.py [hunk 2]
```diff
@@ -34,13 +34,16 @@
 PY_FILE = "test_docs/test_agent_vs_aea/agent_code_block.py"
 
 
+@pytest.mark.flaky(
+    reruns=MAX_FLAKY_RERUNS
+)  # TODO: check why test_run_agent raises permission error on file on windows platform!
```
Comment (DavidMinarsch) [inline tests/test_docs/test_agent_vs_aea/test_agent_vs_aea.py:39]: Reruns need to happen on class level; otherwise no proper setup

**Pull request merged**